src/component/Calculator.js

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from "react";
+import { operators } from "../const";
+import DigitButtons from "./DigitButtons";
+import ModifierButtons from "./ModifierButtons";
+import OperationButtons from "./OperationButtons";
+import "../css/index.css";
+
+// key는 어떤 값으로 지정할까?
+// total은 클릭 영역이 아님
+// 기능 분리하기
+
+export default function Calculator() {
+  const [calc, setCalc] = useState("0");
+
+  const getResult = (e) => {
+    let replace_str = calc.replace(/X/gi, "*");
+    if (isNaN(eval(replace_str))) return setCalc("0");
+    if (eval(replace_str) == Infinity) return setCalc("오류");
+    setCalc(eval(replace_str).toFixed(0));
+  };
+
+  const setResult = (e) => {
+    const inputText = e.target.innerText;
+    const className = e.target.className;
+    if (
+      className != "digit" &&
+      className != "modifier" &&
+      className != "operation"
+    )
+      return;
+
+    if (inputText == "AC") return setCalc("0");
+
+    // 연산자만, 연산자 연속 입력 불가능
+    const lastChar = calc.slice(-1);
+    if (operators.includes(lastChar) && isNaN(inputText)) return;
+
+    if (calc == "0" || calc == "오류") {
+      if (e.target.className == "operation") return;
+      setCalc("");
+    }
+
+    // 세 자리 수 이상 불가능
+    if (Number((calc + inputText).slice(-4)) >= 1000) {
```

Comment:
주석으로 표기해주실게 아니라 영어를 사용하며 명시적으로 변수에 담아서 넣어보실래요?!

Reply:
매직넘버가 될 수 있는 부분같아 보여서
`1000`을 `MAX_NUMBER`로 변경 후 const 변수 파일로 분리했습니다 👍🏻

```suggestion
    if (Number((calc + inputText).slice(-4)) >= MAX_NUMBER) {
```